homeassistant/components/statistics/sensor.py

```diff
@@ -296,15 +297,18 @@ def __init__(
         )
         self._state_characteristic: str = state_characteristic
         self._samples_max_buffer_size: int | None = samples_max_buffer_size
-        self._samples_max_age: timedelta | None = samples_max_age
+        if samples_max_age:
+            self._samples_max_age: float | None = samples_max_age.total_seconds()
+        else:
+            self._samples_max_age = None
         self._precision: int = precision
         self._percentile: int = percentile
         self._value: StateType | datetime = None
         self._unit_of_measurement: str | None = None
         self._available: bool = False
 
         self.states: deque[float | bool] = deque(maxlen=self._samples_max_buffer_size)
-        self.ages: deque[datetime] = deque(maxlen=self._samples_max_buffer_size)
+        self.ages: deque[float] = deque(maxlen=self._samples_max_buffer_size)
```

Comment:
Performance-wise this change makes sense, but the meaning of the float became unclear now. Did you consider renaming `ages` to something speaking?

Reply:
I think they are still ages regardless of how they are stored